src/sentry/models/grouphashmetadata.py

```diff
@@ -78,7 +78,7 @@ class GroupHashMetadata(Model):
     seer_model = models.CharField(null=True)
     # The `GroupHash` record representing the match Seer sent back as a match (if any)
     seer_matched_grouphash = FlexibleForeignKey(
-        "sentry.GroupHash", related_name="seer_matchees", on_delete=models.DO_NOTHING, null=True
+        "sentry.GroupHash", related_name="seer_matchees", on_delete=models.SET_NULL, null=True
```

Comment:
lgtm, there shouldn't be too many `GroupHashMetadata` per `GroupHash`, correct?

Reply:
It's technically possible for it to be a large number, but most often it's not. 

Also, I assume (?) that the null setting happens only when `delete` is called on the linked object, yes? If so, then in cases where there's been no manual unmerge (the 99.9% case), all of the `GroupHashMetadata` records pointing at the `GroupHash` instance will already have been deleted by the time we get to the null setting, as they'll belong to sibling `GroupHash` records of the `GroupHash` record in question. (See point 3 in the first list in the PR description.) So only in the 0.1% case will there even be any remaining `GroupHashMetadata` records on which to update the `seer_matched_grouphash` value.